services/motion/builtin/builtin.go

```diff
@@ -231,25 +235,149 @@ func (ms *builtIn) MoveOnGlobe(
 	kinematicsOptions.GoalRadiusMM = math.Min(motionCfg.PlanDeviationM*1000, 3000)
 	kinematicsOptions.HeadingThresholdDegrees = 8
 
-	plan, kb, err := ms.planMoveOnGlobe(
-		ctx,
-		componentName,
-		destination,
-		movementSensorName,
-		obstacles,
-		kinematicsOptions,
-		extra,
-	)
-	if err != nil {
-		return false, fmt.Errorf("error making plan for MoveOnMap: %w", err)
-	}
+	movementSensor, ok := ms.movementSensors[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	var planMu sync.Mutex
+	var currentPlan [][]referenceframe.Input = nil
+	// TODO: if we disentangle creation of the kinematic base from the plan call we make we don't need this and can do this once
+	var kinematicBase kinematicbase.KinematicBase = nil
+
+	var positionMu sync.Mutex
+	// TODO: this has got to become a geo Point eventually but testing gets hard
+	var currentPosition *geo.Point
+
+	var replanFlag atomic.Bool
```

Comment:
I am having some difficulty picturing exactly how the replanFlag would be flipped? would it be the job of `fn` within `startPolling`?
I think we might want to add a mutex on the replanFlag because what if both position polling and obstacle detection try to write to its value?

Reply:
That's the magic of `atomic`, no mutex necessary to get the same behavior. :)

You are correct, the polling `fn`s would be responsible for flipping that.

Reply:
woahhhhhh that is magic! ✨

Reply:
as cool as atomics are, they aren't actually what we want here I think.  Channels let us wait on interrupts rather than polling the atomic (and effectively busy waiting), so my latest change switches to a channel based implementation